streamlit_drawable_canvas/frontend/src/lib/polypath.ts

```diff
@@ -0,0 +1,116 @@
+import { fabric } from "fabric"
+import FabricTool, { ConfigureCanvasProps } from "./fabrictool"
+
+class PolypathTool extends FabricTool {
+  isMouseDown: boolean = false
+  fillColor: string = "#ffffff"
+  strokeWidth: number = 10
+  strokeColor: string = "#ffffff"
+  currentLine: fabric.Line = new fabric.Line()
+  currentPath: fabric.Path = new fabric.Path()
+  _pathString: string = "M "
+
+  configureCanvas({
+    strokeWidth,
+    strokeColor,
+    fillColor,
+  }: ConfigureCanvasProps): () => void {
+    this._canvas.isDrawingMode = false
+    this._canvas.selection = false
+    this._canvas.forEachObject((o) => (o.selectable = o.evented = false))
+
+    this._canvas.fireRightClick = true
+    this._canvas.stopContextMenu = true
+
```

Comment:
As a side note: The `this._canvas.stopContextMenu = true` seems a reasonable setting in canvas for me. If there is no any use case which need user to bring up the context menu in canvas by right click, then maybe it's a good idea to apply this setting to all the drawing functions?

Reply:
I agree, good job for finding this parameter. You can directly set it in the initialization in `DrawableCanvas.tsx`:

```js
const [canvas, setCanvas] = useState(new fabric.Canvas(""), {
  stopContextMenu: true, // <--  prevent context menu from showing
})
```

Reply:
I just tried to add the setting in `DrawableCanvas.tsx` as the following (I have to put the setting within `new fabric.Canvas()` or error occurs) but it has no effect on stopping the context menu from showing. Do you have any ideas?

```
 52   const [canvas, setCanvas] = useState(
 53     new fabric.Canvas("", {
 54       stopContextMenu: true,
 55     })
 56   )
```

Reply:
Ah yeah that's odd...wonder why this is marked as functional in a Stack Overflow and the option is in the `ICanvasOptions`, should have tested sorry. I'll check the fabricjs issues.

For me the following works

```js
/**
   * State initialization
   */
  const [canvas, setCanvas] = useState(new fabric.Canvas(""))
  canvas.stopContextMenu = true
```